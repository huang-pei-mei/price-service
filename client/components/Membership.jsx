import React from 'react';

const Membership = () => (
  <div id='membership'>
    <div id='premiumPlus' className='priceBody'>
      Included with <span id='premium'>premium</span><span id='plus'>plus</span>
    </div>
    <button className='priceButton priceBody' id='memberEnroll'>Try for $0.00</button>
    <div id='memberBenefits' className='priceBody'>
      <ul>
        <li>1 audiobook of your choice.</li>
        <li>Stream or download thousands of included titles.</li>
        <li>$14.95 a month after 30 day trial. Cancel anytime.</li>
      </ul>
    </div>
  </div>
);

export default Membership;