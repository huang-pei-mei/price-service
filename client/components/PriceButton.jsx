import React from 'react';

const PriceButton = (props) => (
  <div id='price'>
    <button className='priceButton' id='bookBuy'>Buy for ${props.price ? props.price.toFixed(2) : ''}</button>
  </div>
);

export default PriceButton;