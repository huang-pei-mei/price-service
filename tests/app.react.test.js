import React from 'react';
import { shallow } from 'enzyme';
import App from '../client/app.jsx';
import Membership from '../client/components/Membership.jsx';
import PriceButton from '../client/components/PriceButton.jsx';

describe('App', () => {
  beforeEach(() => {
    fetchMock.doMock();
  });

  it('should render my component', () => {
    const component = shallow(<App />);

    expect(component.getElements()).toMatchSnapshot();
  });

  it('should initialize state with currentBook as an empty object', () => {
    const component = shallow(<App />);

    expect(component.state('currentBook')).toBeDefined();
  });

  it('should render Membership and PriceButton components', () => {
    const component = shallow(<App />);
    expect(component.contains(<Membership />)).toBe(true);

    expect(component.contains(<PriceButton />)).toBe(true);
  });

  it('should call componentDidMount', () => {
    const component = shallow(<App />);
    const instance = component.instance();

    jest.spyOn(instance, 'componentDidMount');

    instance.componentDidMount();

    expect(instance.componentDidMount).toHaveBeenCalledTimes(1);

  });

  it('should call setParam', () => {
    const component = shallow(<App />);
    const instance = component.instance();

    jest.spyOn(instance, 'setParam');

    instance.componentDidMount();

    expect(instance.setParam).toHaveBeenCalledTimes(1);
  });

  it('should call getPrice', () => {
    const component = shallow(<App />);
    const instance = component.instance();

    jest.spyOn(instance, 'getPrice');

    instance.componentDidMount();

    expect(instance.getPrice).toHaveBeenCalledTimes(1);
  });


});