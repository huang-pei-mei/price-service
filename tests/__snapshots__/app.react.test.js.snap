// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render my component 1`] = `
Array [
  <div
    id="app"
  >
    <Membership />
    <PriceButton />
  </div>,
]
`;
